src/async-status-indicator/index.js

```diff
@@ -1,2 +1,4 @@
-import AsyncStatusIndicator from './AsyncStatusIndicator';
-export default AsyncStatusIndicator;
\ No newline at end of file
+import AsyncStatusIndicator, { Props as AsyncStatusIndicatorProps, AsyncStatusIndicatorState } from './AsyncStatusIndicator';
+
+export { AsyncStatusIndicatorProps, AsyncStatusIndicatorState };
+export default AsyncStatusIndicator;
```

Comment:
NOTE: I'm not 100% sure but I think you can export everything with this one liner:

```js
export default, { Props as AsyncStatusIndicatorProps, AsyncStatusIndicatorState } from './AsyncStatusIndicator';
```